nntrainer/tensor/task_executor.h

```diff
@@ -0,0 +1,162 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2022 Jiho Chu <jiho.chu@samsung.com>
+ *
+ * @file   task_executor.h
+ * @date   04 Nov 2022
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Jiho Chu <jiho.chu@samsung.com>
+ * @bug    No known bugs except for NYI items
+ * @brief  Task executor class
+ *
+ */
+
+#ifndef __TASK_EXECUTOR_H__
+#define __TASK_EXECUTOR_H__
+
+#include <atomic>
+#include <functional>
+#include <future>
+#include <map>
+#include <memory>
+#include <unistd.h>
+
+#include <nntrainer_error.h>
+#include <task.h>
+
+namespace nntrainer {
+
+/**
+ * @class   TaskExecutor
+ * @brief   task executor class
+ */
+class TaskExecutor {
+public:
+  enum CompleteStatus {
+    SUCCESS,
+    FAIL_CANCEL,
+    FAIL_TIMEOUT,
+    FAIL,
+  };
+
+  using CompleteCallback =
```

Comment:
Could you add more explanation as an comment?

Reply:
thanks. it added.